clients/include/blas2/testing_hpr.hpp

```diff
@@ -37,6 +37,68 @@ inline void testname_hpr(const Arguments& arg, std::string& name)
     hipblasHprModel{}.test_name(arg, name);
 }
 
+template <typename T>
+void testing_hpr_bad_arg(const Arguments& arg)
+{
+    using U           = real_t<T>;
+    bool FORTRAN      = arg.api == hipblas_client_api::FORTRAN;
+    auto hipblasHprFn = FORTRAN ? hipblasHpr<T, U, true> : hipblasHpr<T, U, false>;
+
+    for(auto pointer_mode : {HIPBLAS_POINTER_MODE_HOST, HIPBLAS_POINTER_MODE_DEVICE})
+    {
+        hipblasLocalHandle handle(arg);
+        CHECK_HIPBLAS_ERROR(hipblasSetPointerMode(handle, pointer_mode));
+
+        hipblasFillMode_t uplo   = HIPBLAS_FILL_MODE_UPPER;
```

Comment:
Well that is fine too, do you ever see potential in testing an invalid enum with bad arg?  If you just use arg.uplo instead of a constant this could allow it.  Only concern was putting something in the yaml you expect it may be used, so that fact that it isn't may just lead poor understanding of the test set, or test coverage.

Reply:
I don't think I see an advantage using yaml for things like enums; invalid enums are just tested by coding it in the testing_*.hpp file right now which I think is better. Could just take the regular enum from the yaml file like you say not sure there's much of an advantage.

Not just the enums but things like sizes in the yaml just don't make much sense for the bad_arg tests which is why I made the change. Seeing a test name like `gemv_bad_arg_a_type_f32_r` makes more sense than having a bunch of params that don't really apply to that test.

Reply:
Good I see they get removed from the yaml